### PR TITLE
feat(build metrics): build metrics UI

### PR DESCRIPTION
**Pull request by DekusDenial**
UI for alpha/private beta release:

There is no exposed UI cta to link to the metrics page, you will have to append `/metrics` to pipeline page, for example: `/pipelines/123/metrics`

*note: If I were a reviewer, I will not struggle too much on making little things configurable. Esp with chart style which overrides a number of style from default c3*

**YES:**
- preset time range + custom start & end time selection
- events and builds breakdown charts
- trendline toggle for events chart
- custom tooltip for revealing data details
- dynamic responsive tick label generation for x-axis
- zoom-in by dragging across chart
- zoom-in and zoom-reset synchronization between charts (except during individual sub bar toggle)
- default to **1wk** (cuz i wanna see a good chunk of data points)

**NO:**
- custom legend (just little cosmetic with too much override, even more than tooltip)
- sub bar visibility toggle vs zoom level (once toggle, zoom level will reset, issue in c3)
- third step breakdown chart
- aggregation

Screen Shot 2019-03-26 at 5 35 05 PM

related: https://github.com/screwdriver-cd/screwdriver/issues/1412

### CODE REVIEW

File: app/components/chart-c3/component.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      bar: this.get('bar'),
+      bindto: this.element,
+      color: this.get('color'),
+      data: this.get('data'),
+      grid: this.get('grid'),
+      legend: this.get('legend'),
+      oninit: this.get('oninit'),
+      onrendered: this.get('onrendered'),
+      onresized: this.get('onresized'),
+      padding: { top: 30 },
```
Comment (DekusDenial) [inline app/components/chart-c3/component.js]: this is there cuz overriding c3 y-axis label is hacky and risk clipping the text so i moved the label out of d3 and needed preserve pixels for the overridden legend as well

File: app/pipeline/metrics/template.hbs [hunk 2]
```diff
@@ -0,0 +1,127 @@
+{{#if session.isAuthenticated}}
+  {{#if (not session.data.authenticated.isGuest)}}
```
Comment (jithin1987) [inline app/pipeline/metrics/template.hbs]: why restrict `Guest` ?

File: app/pipeline/model.js [hunk 3]
```diff
@@ -18,6 +18,7 @@ export default DS.Model.extend({
   jobs: DS.hasMany('job', { async: true }),
   secrets: DS.hasMany('secret', { async: true }),
   tokens: DS.hasMany('token', { async: true }),
+  metrics: DS.hasMany('metric', { async: true }),
```
Comment (jithin1987) [inline app/pipeline/model.js]: does setting `metrics` at pipeline model result in fetching this in all nested pages under `pipeline`?

File: app/styles/chart.scss [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  font-weight: bold;
+  margin-right: 10px;
+
+  button {
+    background: #eff4ff;
+    padding: 5px 10px;
+    line-height: 1em;
+    border: none;
+    border-radius: 3px;
+    font-weight: norma;
```
Comment (jithin1987) [inline app/styles/chart.scss]: `normal` ?

File: app/styles/chart.scss [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+.chart-c3 {
+    margin-bottom: 15px;
+    padding: 15px 15px 10px;
+    background: #f8f8f8;
+    border: 1px solid #dadada;
+    border-radius: 5px;
+    position: relative;
+    // box-shadow: inset 0 0 10px -5px #999;
```
Comment (jithin1987) [inline app/styles/chart.scss]: remove ?

File: app/pipeline/metrics/template.hbs (hunk 2 above)
Comment (DekusDenial) [reply app/pipeline/metrics/template.hbs]: cuz it's for complete alpha/private beta.
But i can relax it then

File: app/pipeline/model.js (hunk 3 above)
Comment (DekusDenial) [reply app/pipeline/model.js]: no. fetched only when explicitly fetched in code

**Pull request merged**